src/http/proxy_wasm/ngx_http_proxy_wasm_dispatch.c

```diff
@@ -606,7 +621,24 @@ ngx_http_proxy_wasm_dispatch_request(ngx_http_proxy_wasm_dispatch_t *call)
 
     b->last = ngx_cpymem(b->last, ngx_http_proxy_wasm_host,
                          sizeof(ngx_http_proxy_wasm_host) - 1);
+
+#if (NGX_HAVE_UNIX_DOMAIN)
+    static ngx_str_t uds = ngx_string("unix:");
+    static ngx_str_t localhost = ngx_string("localhost");
+
+    if (call->authority.len > uds.len
+        && ngx_str_eq(call->authority.data, uds.len, uds.data, uds.len))
+    {
+        b->last = ngx_cpymem(b->last, localhost.data, localhost.len);
+
+    } else {
+        b->last = ngx_cpymem(b->last, call->authority.data,
```

Comment:
There are better ways to use preprocessor directives that would avoid duplicating this line.